modules/persistenceService/src/main/scala/subsPersistenceLayer/SubscriptionServicePersistenceLayer.scala

```diff
@@ -43,10 +54,6 @@ object SubscriptionServiceQuery {
     val sql = "insert into subscriptions(user_id, repository_id) values(?,?)"
     Update[SubscriptionInfo](sql)
   }
-
-  def delete(id: Id): Update[PostSubscriptionData] = {
-    val sql =
-      s"delete from subscriptions where id = ${id.id} and repository_id in (select * from repositories where owner = ? and repository = ?)"
-    Update[PostSubscriptionData](sql)
-  }
```

Comment:
I still don't fully understand why this was failing because the signature of `Update` takes a String as the first param and  `s""` is a string. I guess it is the fact that an `id = ${id.id}` was being used in the string but if anyone could explain I will be grateful.

```
  def apply[A](sql0: String, pos0: Option[Pos] = None)(
      implicit W: Write[A], logHandler0: LogHandler = LogHandler.nop
    ): Update[A] = ...
```
Batch update Ref
 - https://tpolecat.github.io/doobie/docs/07-Updating.html
 
<img width="861" alt="Screenshot 2021-08-19 at 21 15 57" src="https://user-images.githubusercontent.com/37377831/130137850-55bee82a-bd7e-48d7-92c1-8e8a627dd91c.png">

Reply:
What was the error @SarpongAbasimi?

Reply:
Thank for the review @fedefernandez please if I had something like this.

```
      def delete(userId: UserId, subs: Subs): ConnectionIO[Unit] = for {
        _ <- subs.appSubscriptions.traverse { data =>
          SubsAppServiceQuery
            .delete(userId)
            .updateMany(
              List(
                (data.name, data.age)
              )
            )
            .void
        }
      } yield ()
    }
```
With delete being 
```
 def delete(userId: UserId): Update[(Name, Age)] = {
    val sql =
      s"""
         |delete from sub where user_id = ${userId.id} and 
         |index_id in 
         |(select index_id from friends where name=? and age=?)""".stripMargin
    Update[(Name, Age)](sql)
  }

```

The error was in the  line of 

```
delete from subs where user_id = 8c878e6f-ee13-4a37-a208-7510c2638945 and
index_id in
(select index_id from friends where name='Uzokoolu' and age='aduasa')
 was aborted: ERROR: syntax error at or near "c878e6f"
```

Reply:
can you try quoting the id:
```diff
- delete from sub where user_id = ${userId.id} and
+ delete from sub where user_id = '${userId.id}' and
```

Reply:
Oh, the problem is because you're passing a raw SQL, you need to manually scape the strings

```scala
  def delete(id: Id): Update[PostSubscriptionData] = {
    val sql =
      s"delete from subscriptions where id = '${id.id}' and repository_id in (select * from repositories where owner = ? and repository = ?)"
    Update[PostSubscriptionData](sql)
  }
```

Reply:
Ahhh okay, that explains it. Thank you very much @fedefernandez

Reply:
But how did you figure this out please @fedefernandez?
Just curious

Reply:
> can you try quoting the id:
> 
> ```diff
> - delete from sub where user_id = ${userId.id} and
> + delete from sub where user_id = '${userId.id}' and
> ```

Thank you ☺️

Reply:
@SarpongAbasimi saw a lot of times in the past 😅 , is trying to read the id as a variable and stops on the `-`
```
was aborted: ERROR: syntax error at or near "c878e6f"
```